.github/workflows/build.yml

```diff
@@ -0,0 +1,61 @@
+name: Test build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build:
+    name: Test build on Linux
+    strategy:
+      fail-fast: false
+      matrix:
+        luaVersion: [ "5.4", "5.3", "5.2", "5.1", "luajit", "luajit-openresty"]
+        platform: [ "ubuntu-20.04", "macos-11" ] # "windows-2022" not supported by gh-actions-lua
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup ‘lua’
+        uses: leafo/gh-actions-lua@v9
+        with:
+          luaVersion: ${{ matrix.luaVersion }}
+      - name: Setup ‘luarocks’
+        uses: leafo/gh-actions-luarocks@v4
+      - name: Make and install
+        run: |
+          export DEBUG=DEBUG
+          luarocks make -- luasocket-scm-3.rockspec
+      - name: Run regression tests
+        run: |
+          cd test
+          lua hello.lua
+          lua testsrvr.lua > /dev/null &
+          lua testclnt.lua
+          lua stufftest.lua
+          lua excepttest.lua
+          lua test_bind.lua
+          lua test_getaddrinfo.lua
+          lua ltn12test.lua
+          lua mimetest.lua
+          lua urltest.lua
+          lua test_socket_error.lua
+          # lua dicttest.lua
+          # lua ftptest.lua
+          # lua hello.lua
+          # lua httptest.lua
+          # lua smtptest.lua
+          # lua testmesg.lua
+          # lua testsupport.lua
+          # lua tftptest.lua
+          # lua udpconnectclnt.lua
+          # lua udpconnectsrvr.lua
+          # lua unixdgramclnt.lua
+          # lua unixdgramsrvr.lua
+          # lua unixstreamclnt.lua
+          # lua unixstreamsrvr.lua
+          # lua utestclnt.lua
+          # lua utestsrvr.lua
```

Comment:
Why are these here and commented out?

Reply:
Because I was playing around with files that were in tests but not part of the current suite. I'll remove them before taking this out of draft mode.